src/beta_inc.jl

```diff
@@ -0,0 +1,876 @@
+                end
+                b0[m] = r*a0[m] + bsum/m
+            end
+            c[i] = b0[i]/(i+1.0)
+            dsum = 0.0
+            for j = 1: i-1
+                imj = i - j
+                dsum += d[imj]*c[j]
+            end
+            d[i] = -(dsum + c[i])
+        end
+
+        j0 = e1*znm1 + (n - 1.0)*j0
+        j1 = e1*zn + n*j1
+        znm1 *= z²
+        zn *= z²
+        w *= w0
+        t0 = d[n]*w*j0
+        w *= w0
+        t1 = d[n+1]*w*j1
+        sm += (t0 + t1)
+        if (abs(t0) + abs(t1)) <= epps*sm
+            break
+        end
+    end
+
+    u = exp(-bcorr(a,b))
+    return e0*t*u*sm
+end        
+
+"""
+    bgrat(a,b,x,y,w,epps)
+
+Evaluation of I_{x}(a,b) when b < min(epps,epps*a) and x <= 0.5 using asymptotic expansion.
+It is assumed a >= 15 and b <= 1, and epps is tolerance used.
+"""
+function bgrat(a::Float64, b::Float64, x::Float64, y::Float64, w::Float64, epps::Float64)
+    c = zeros(31)
+    d = zeros(31)
+    bm1 = b - 1.0
+    nu = a + 0.5*bm1
+    if y > 0.375
+        lnx = log(x)
+    else
+        lnx = log1p(-y)
+    end
+    z = -nu*lnx
+    if b*z == 0.0
+        return error("expansion can't be computed")
+    end
+
+    # COMPUTATION OF THE EXPANSION
+    #SET R = EXP(-Z)*Z**B/GAMMA(B)
+    r = b*(1.0 + SpecialFunctions.rgamma1pm1(b))*exp(b*log(z))
+    r *= exp(a*lnx)*exp(0.5*bm1*lnx)
+    u = loggammadiv(b,a) + b*log(nu)
+    u = r*exp(-u)
+    if u == 0.0
+        return error("expansion can't be computed")
+    end
+    (p, q) = gamma_inc(b,z,0)
+    v = 0.25*(1.0/nu)^2.0
+    t2 = 0.25*lnx*lnx
+    l = w/u
+    j = q/r
+    sm = j
+    t = 1.0
+    cn = 1.0
+    n2 = 0.0
+    for n = 1:30
+        bp2n = b + n2
+        j = (bp2n*(bp2n + 1.0)*j + (z + bp2n + 1.0)*t)*v
+        n2 += 2.0
+        t *= t2
+        cn /= n2*(n2 + 1.0)
+        c[n] = cn
+        s = 0.0
+        if n != 1
+            nm1 = n -1
+            coef = b - n
+            for i = 1:nm1
+                s += coef*c[i]*d[n-i]
+                coef += b
+            end
+        end
+        d[n] = bm1*cn + s/n
+        dj = d[n] * j
+        sm += dj
+        if sm <= 0.0
+            return error("expansion can't be computed")
+        end
+        if abs(dj) <= epps*(sm+l)
+            break
+        end
+    end
+    return w + u*sm
+end 
+
+
+
+function fpser(a::Float64, b::Float64, x::Float64, epps::Float64)
+    ans = 1.0
+    if a > 1.0e-3 * epps
+        ans = 0.0
+        t = a*log(x)
+        if t < exparg_n
+            return ans
+        end
+        ans = exp(t)
+    end
+    ans *= b/a
+    tol = epps/a
+    an = a + 1.0
+    t = x
+    s = t/an
+    an += 1.0
+    t *= x
+    c = t/an
+    s += c
+    while abs(c) > tol 
+        an += 1.0
+        t *= x
+        c = t/an
+        s += c
+    end
+    ans *= (1.0 + a*s)
+    return ans
+end
+
+#A <= MIN(EPS,EPS*B), B*X <= 1, AND X <= 0.5.,  A is small
+
+function apser(a::Float64, b::Float64, x::Float64, epps::Float64)         
```

Comment:
Do you know which function this corresponds to from the paper? Is it a variant of `bpser`?

Reply:
Even I didn't find this in the paper. Yes, it is a variant of `bpser` as far as I had understood.

Reply:
Lets call it `beta_inc_power_series1` or something like that until we can figure out a better name.